app/src/features/sources/SourceForm.tsx

```diff
@@ -0,0 +1,120 @@
+import React from "react";
+import { useTranslation } from "react-i18next";
+
+import Form from "@arkhn/ui/lib/Form/Form";
+import {
+  Box,
+  Button,
+  CircularProgress,
+  makeStyles,
+  Typography,
+} from "@material-ui/core";
+
+import { Source } from "services/api/generated/api.generated";
+import { FormInputProperty } from "@arkhn/ui/lib/Form/InputTypes";
+import { TFunction } from "i18next";
+import {
+  useCreateSourceMutation,
+  useUpdateSourceMutation,
+} from "services/api/api";
+
+type SourceFormData = {
+  name: string;
+};
+
+type SourceFormProps = {
+  source?: Source;
+  submit?: (source: Source) => void;
+};
+
+const inputs: (t: TFunction) => FormInputProperty<SourceFormData>[] = (t) => [
+  {
+    type: "text",
+    name: "name",
+    label: t("sourceName"),
+    variant: "outlined",
+    validationRules: { required: true },
+  },
+];
+
+const useStyles = makeStyles((theme) => ({
+  formContainer: {
+    marginBlock: theme.spacing(3),
+    minWidth: 400,
+  },
+  title: {
+    marginLeft: theme.spacing(3),
+    fontWeight: "bold",
+  },
+  button: {
+    marginLeft: theme.spacing(3),
+    textTransform: "none",
+    width: "auto",
+    minWidth: 150,
+  },
+}));
+
+const SourceForm = ({ source, submit }: SourceFormProps) => {
+  const { t } = useTranslation();
+  const classes = useStyles();
+
+  const [createSource, createSourceData] = useCreateSourceMutation();
+  const [updateSource, updateSourceData] = useUpdateSourceMutation();
```

Comment:
same for `updateSourceData`, I think you could use something like `{isLoading: isSourceUpdateLoading}`